### PR TITLE
Implement /api/derive/session/progress call

### PR DESCRIPTION
**Pull request by aspekt112**
This API call allow us to obtain much more info about current session and era:
```
/api/derive/session/progress
{"result":{"activeEra":"0x00000a29","activeEraStart":1628594874004,"currentEra":"0x00000a29","currentIndex":"0x00003a35","validatorCount":"0x00000384","eraLength":"0x00000e10","isEpoch":true,"sessionLength":"0x0000000000000258","sessionsPerEra":"0x00000006","eraProgress":"0x000007fb","sessionProgress":"0x000000f3"}}
```

### CODE REVIEW

**Comment by dillu24** [comment]
Hi @aspekt112 , thanks a lot for your contribution! We will give it a look hopefully soon.

**Comment by aspekt112** [comment]
@dillu24 Hello! How soon can you review these changes?

**Comment by dillu24** [comment]
Later this week hopefully

**Comment by dillu24** [comment]
@aspekt112 - Sorry for the delay, will start reviewing now.

File: doc/DESIGN_AND_FEATURES.md [hunk 1]
```diff
@@ -64,11 +64,13 @@ The API Server works as follows:
...
 | `/api/query/staking/validators` | `websocket`, `account_id` | Preferences of the validator whose stash is the specified `account_id` (validator commission and blocked status). |
+| `/api/query/system/account` | `websocket`, `account_id` |  None | The full account information for a particular account ID |
 | `/api/query/system/events` | `websocket` | `block_hash` | `events` that happened in the specified `block hash`, or in the latest block if the block hash is not specified |
 | **Custom** | | | |
 | `/api/custom/getSlashAmount` | `websocket`, `account_address` | `block_hash` | The `balance slashed` (if any) of the specified `account address` in the specified `block hash`, or in the latest block if the block hash is not specified |
 | **Derive** | | | |
 | `/api/derive/staking/validators` | `websocket` | None | `nextElected` - List of `validators` which will be active in the `next session` and `validators` - List of `validators` which are currently active |
+| `/api/derive/session/progress` | `websocket` | None | retrieves all the `session` and `era` query and calculates specific values on it as the length of the session and eras |
```
Comment (dillu24) [inline doc/DESIGN_AND_FEATURES.md]: Change the description to `Retrieves information about the current session`

**Review by dillu24 — CHANGES_REQUESTED** [review]
Good Work! Require only few minor changes:

1. Change the endpoint description as described in one of the threads.
2. Add the following entry to `doc/CHANGELOG.md`'s `Unreleased Section`:
* Added `session/progress`. This returns information about the current session.
3. Since you have already pulled from base's code, we will wait for PR #36 to be merged to our repo due to some requested changes. Once #36 is merged, please pull from master to have the latest code-base, and push to your branch.

File: demos/api_demo.py [hunk 2]
```diff
@@ -257,6 +257,11 @@
                  params={'websocket': websocket})
 print(r.text)
 
+print('/api/derive/session/progress')
```
Comment (DanielMagro97) [inline demos/api_demo.py]: Kindly move this before `/api/derive/staking/validators`, to keep things in alphabetical order

File: doc/DESIGN_AND_FEATURES.md (hunk 1 above)
Comment (DanielMagro97) [reply doc/DESIGN_AND_FEATURES.md]: Also, same as my other comment, kindly put this above `/api/derive/staking/validators`

File: doc/DESIGN_AND_FEATURES.md [hunk 3]
```diff
@@ -64,11 +64,13 @@ The API Server works as follows:
 | `/api/query/staking/payee` | `websocket`, `account_id` | None | Reward destination address assigned to the stash with the specified `account_id`. Returns json with different keys depending on the type of reward destination: `staked: null` - when reward destination is a stash, `controller: addr` - when reward destination is a controller, and `result: addr` when reward destination is any other address. |
 | `/api/query/staking/unappliedSlashes` | `websocket` | `era_index` | List of slashed validators (unapplied) in the specified era, or the current era if an era index is not specified. |
 | `/api/query/staking/validators` | `websocket`, `account_id` | Preferences of the validator whose stash is the specified `account_id` (validator commission and blocked status). |
+| `/api/query/system/account` | `websocket`, `account_id` |  None | The full account information for a particular account ID |
```
Comment (DanielMagro97) [inline doc/DESIGN_AND_FEATURES.md]: The full account information for the specified `account_id`

File: src/interface/substrate_derive.js [hunk 4]
```diff
@@ -10,6 +10,14 @@ async function getStakingValidators(api) {
     );
 }
 
+// Session
```
Comment (DanielMagro97) [inline src/interface/substrate_derive.js]: Same as before, kindly put Session before Staking

File: src/interface/substrate_derive.js [hunk 5]
```diff
@@ -20,6 +28,13 @@ module.exports = {
                 } catch (e) {
                     return {'error': e.toString()};
                 }
+            // Session
```
Comment (DanielMagro97) [inline src/interface/substrate_derive.js]: Same here

File: src/server.js [hunk 6]
```diff
@@ -1189,6 +1222,36 @@ async function startPolkadotAPI() {
                 {'error': e.toString()});
         }
     });
+
+    // Session
```
Comment (DanielMagro97) [inline src/server.js]: Again kindly put this before the Derive Staking one to preserve the alphabetical order

**Review by DanielMagro97 — CHANGES_REQUESTED** [review]
Just a few minor comments for rewording and reordering of functions

**Comment by aspekt112** [comment]
@dillu24 @DanielMagro97 can you review fixes?

**Comment by dillu24** [comment]
@aspekt112 - Sorry for the delay, we will be reviewing this week for sure.

**Comment by aspekt112** [comment]
@dillu24 @DanielMagro97 Please review another commit. Last runtime upgrade made major API changes

**Comment by dillu24** [comment]
@aspekt112 .. Waiting for #36 to be merged. Once it is merged please merge from master into this branch so that the latest changes are reflected.

**Pull request closed without merging**